### PR TITLE
rework how the manager watches PIDs

### PR DESCRIPTION
**Pull request by poettering**

### CODE REVIEW

**Comment by poettering** [comment]
Force pushed a new rebased version now,

**Comment by sourcejedi** [comment]
https://github.com/systemd/systemd/pull/7855/files/c1e83af84eea52c22f40ae8f34b1c5a32d94a90d..8895eb7815de9931e043c2a3bd3e6750bee47719#diff-f71635b4a3a6151182c1ae585fb5e3ad

https://github.com/systemd/systemd/blob/8895eb7815de9931e043c2a3bd3e6750bee47719/src/core/service.c#L2593

> hmm, `_debug`. I see parse failures are all logged as _debug, but there's a couple of _error's for things that look like a clear ENOMEM. This PR part seems to be related to #7860. After #7860, unit_watch_pid() should stop failing if too many different units watch the same pid, and should only fail for ENOMEM, right? But I [still] didn't find a change of this line in #7860. Should #7860 upgrade this message from _debug to log_unit_error_errno()?

**Comment by poettering** [comment]
Well, our philosophy regarding log levels was to use LOG_ERR and above only if we actually let an operatoin fail, but use LOG_WARN (or LOG_DEBUG) if we proceed and ignore the effect. We also had the philosophy that deserializing the serialization state should never result in errors logged above LOG_DEBUG, since we do inter-version updates that way and we never really know what we operate on, and such errors are not actionable anyway.

So, yeah, dunno, which log level to use is not that obvious a choice I figure. Given that previously we didn't log here at all I decided to go only one step up and log at debug level only... 

Or to say this differently: we can decide to upgrade logging during serialization, but if we do so, we should really do it all the way for all cases, not just this one...

**Comment by sourcejedi** [comment]
I am being quite nitpicky, but there _are_ LOG_ERR messages in `service_deserialize_item()`.  I interpreted the distinction there as being that parse failures - i.e. errors in the input - are LOG_DEBUG, but the operations which might fail because of ENOMEM, are treated as LOG_ERR.  I.e. when the system is breaking down because it has run out of resources.

I'm assuming that previously, a unit_watch_pid() failure could be argued as a possible input failure, because it would not work if too many units had the same main pid.  After this PR, unit_watch_pid() failures should be a clear sign of ENOMEM.

I suppose the apparent distinction does not account for ENOMEM failures due to input strings which are invalidly long :).

**Comment by sourcejedi** [comment]
Sorry, I got frustrated and deleted my first comment.

I appreciate not necessarily logging an error if the operation as a whole is treated as successful.

However there _are_ a couple of points in the service deserialization where errors were logged already.  Despite the operation being allowed to continue.  Looking at them, I believe these correspond to ENOMEM, i.e. the system is breaking down because of lack of resources.  As opposed to errors with the input, e.g. parse failures.  (Possibly due to a downgrade of systemd).

Previously a failure of unit_watch_pid() appeared ambiguous, because it wouldn't work if too many units were marked as having the same man pid.  But after this PR, the failure appears to be an unambiguous sign of ENOMEM.

So the function reads more consistent if the failure is upgraded to LOG_ERR at this point.

**Comment by poettering** [comment]
So, I just realized that watching the PID during deserialization there is pointless anyway, as service_coldplug() does that anyway later on. Hence, we can just drop the call altogether and make things simpler.

**Comment by yuwata** [comment]
Unfortunately, needs to rebase.

**Comment by poettering** [comment]
already on it

**Comment by poettering** [comment]
force pushed a new version now. ptal!

File: src/core/dbus-cgroup.c [hunk 1]
```diff
@@ -533,7 +533,7 @@ int bus_cgroup_set_property(
         if (streq(name, "MemoryLimitScale"))
                 return bus_cgroup_set_memory_scale(u, name, &c->memory_limit, message, flags, error);
 
-        if (streq(name, "TasksAccountingScale"))
+        if (streq(name, "TasksAccounting"))
```
Comment (yuwata) [inline src/core/dbus-cgroup.c:536]: Ouch! Thank you for finding this.

**Comment by poettering** [comment]
OK, it passed all CI, and @yuwata liked it, let's get this merged!

Thanks for reviewing!

**Pull request merged**